org.openrewrite.java.format.WrappingAndBracesVisitor crashes when tryWithResources.wrap is set to ChopIfTooLong

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using

- OpenRewrite v8.85.0
- Maven plugin v6.42.0
- rewrite-java v8.85.0

## How are you running OpenRewrite?
<!--
Are you using the Maven plugin, Gradle plugin, Moderne CLI, Moderne.io or something else?
Is your project a single module or a multi-module project?

Can you share your configuration so that we can rule out any configuration issues?

Is your project public? If so, can you share a link to it?
Code snippets can also be shared privately via [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).
-->
I am using the Maven plugin, and my project is a multi module project.
```xml
<plugin>
  <groupId>org.openrewrite.maven</groupId>
  <artifactId>rewrite-maven-plugin</artifactId>
  <version>6.42.0</version>
  <executions>
      <execution>
          <id>code-style</id>
          <configuration>
              <exclusions>
                  <exclusion>**/Jenkinsfile</exclusion>
                  <exclusion>**/.m2/**</exclusion>
                  <exclusion>**/resources/**</exclusion>
                  <exclusion>**/target/**</exclusion>
                  <exclusion>**/*.zip</exclusion>
                  <exclusion>**/*.jar</exclusion>
                  <exclusion>**/beans.xml</exclusion>
              </exclusions>
              <skipMavenParsing>true</skipMavenParsing>
              <pomCacheEnabled>false</pomCacheEnabled>
              <exportDatatables>true</exportDatatables>
              <activeRecipes>
                  <recipe>org.openrewrite.java.RemoveUnusedImports</recipe>
                  <recipe>org.openrewrite.java.format.AutoFormat</recipe>
              </activeRecipes>
              <activeStyles>
                  <style>eu.tennet.java.CodingStyle</style>
              </activeStyles>
          </configuration>
      </execution>
  </executions>
</plugin>
```
`rewrite.yml`:
```yaml
---
type: specs.openrewrite.org/v1beta/style
name: eu.tennet.java.CodingStyle
displayName: Java coding style
styleConfigs:
  - org.openrewrite.java.style.WrappingAndBracesStyle:
      hardWrapAt: 170
      tryWithResources:
        wrap: ChopIfTooLong
        alignWhenMultiline: true
```

Command:
`mvn rewrite:dryRun@code-style`

## What is the smallest, simplest way to reproduce the problem?
<!--
Sometimes the logs indicate a recipe stumbled over a particular pattern of code.
If you can share a code snippet that reproduces the issue, that will help us fix it faster.
We also accept [pull requests that merely replicate an issue](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes), as a step up to a full fix.

A code snippet can be something simple like this, or similar for other languages:
-->
```java
class A {
    public String writeEventsAsCSV(List<Event> events) {
        var sw = getWriter();

        try (var printer = new CSVPrinter(sw, csvFormat)) {
            events.forEach(event -> {
                try {
                    printer.printRecord(
                            event.getField1(),
                            event.getField2(),
                            event.getField3().withZoneSameInstant(TIME_ZONE).format(DateTimeFormatter.ofPattern(LOCAL_DATETIME_FORMAT)),
                            event.getField4(),
                            event.getField5(),
                            event.getField6(),
                            event.getField7(),
                            event.getField8(),
                            event.getField9(),
                            event.getField10(),
                            event.getField11(),
                            event.getField12(),
                            event.getField13(),
                            event.getField14()
                    );
                } catch (IOException e) {
                    throw new RuntimeException(String.format("Failed to write event to CSV format '%s'", event), e);
                }
            });
        } catch (Exception e) {
            log.error("Failed to write events to CSV format", e);
        }

        return sw.toString();
    }
}
```

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:6.42.0:dryRun (code-style) on project x: Execution code-style of goal org.openrewrite.maven:rewrite-maven-plugin:6.42.0:dryRun failed: Error while visiting A.java: java.lang.IllegalArgumentException: The child was not found in the sourceFile. Are you sure the passed in cursor's value contains the child and you are not searching for a mutated element in a non-mutated Cursor value?                                       
  org.openrewrite.java.service.SourcePositionService.positionOfChild(SourcePositionService.java:401)                                                                                                                                                                                                                      
  org.openrewrite.java.service.SourcePositionService.columnsOf(SourcePositionService.java:90)                                                                                                                                                                                                                             
  org.openrewrite.java.format.WrappingAndBracesVisitor.shouldWrap(WrappingAndBracesVisitor.java:529)                                                                                                                                                                                                                      
  org.openrewrite.java.format.WrappingAndBracesVisitor.lambda$visitContainer$1(WrappingAndBracesVisitor.java:89)                                                                                                                                                                                                          
  org.openrewrite.internal.ListUtils.map(ListUtils.java:206)                                                                                                                                                                                                                                                              
  org.openrewrite.java.format.WrappingAndBracesVisitor.visitContainer(WrappingAndBracesVisitor.java:84)                                                                                                                                                                                                                   
  org.openrewrite.java.JavaVisitor.visitTry(JavaVisitor.java:1177)                                                                                                                                                                                                                                                        
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:330)                                                                                                                                                                                                                                                   
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:30)                                                                                                                                                                                                                                                    
  org.openrewrite.java.tree.J$Try.acceptJava(J.java:5701)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.tree.J.accept(J.java:55)                                                                                                                                                                                                                                                                           
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:242)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:461)                                                                                                                                                                                                                           
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:48)                                                                                                                                                                                                                            
  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:328)                                                                                                                                                                                                                                                          
  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1350)                                                                                                                                                                                                                                                
  ...                                                                                                                                                                                                                                                                                                                     
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution code-style of goal org.openrewrite.maven:rewrite-maven-plugin:6.42.0:dryRun failed: Error while visiting A.java: java.lang.IllegalArgumentException: The child was not found in the sourceFile. Are you sure the passed in cursor's value contains the child and you are not searching for a mutated element in a non-mutated Cursor value?                                                                                                                                                              
  org.openrewrite.java.service.SourcePositionService.positionOfChild(SourcePositionService.java:401)                                                                                                                                                                                                                      
  org.openrewrite.java.service.SourcePositionService.columnsOf(SourcePositionService.java:90)                                                                                                                                                                                                                             
  org.openrewrite.java.format.WrappingAndBracesVisitor.shouldWrap(WrappingAndBracesVisitor.java:529)                                                                                                                                                                                                                      
  org.openrewrite.java.format.WrappingAndBracesVisitor.lambda$visitContainer$1(WrappingAndBracesVisitor.java:89)                                                                                                                                                                                                          
  org.openrewrite.internal.ListUtils.map(ListUtils.java:206)                                                                                                                                                                                                                                                              
  org.openrewrite.java.format.WrappingAndBracesVisitor.visitContainer(WrappingAndBracesVisitor.java:84)                                                                                                                                                                                                                   
  org.openrewrite.java.JavaVisitor.visitTry(JavaVisitor.java:1177)                                                                                                                                                                                                                                                        
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:330)                                                                                                                                                                                                                                                   
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:30)                                                                                                                                                                                                                                                    
  org.openrewrite.java.tree.J$Try.acceptJava(J.java:5701)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.tree.J.accept(J.java:55)                                                                                                                                                                                                                                                                           
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:242)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:461)                                                                                                                                                                                                                           
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:48)                                                                                                                                                                                                                            
  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:328)                                                                                                                                                                                                                                                          
  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1350)                                                                                                                                                                                                                                                
  ...                                                                                                                                                                                                                                                                                                                     
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.RuntimeException: Error while visiting A.java: java.lang.IllegalArgumentException: The child was not found in the sourceFile. Are you sure the passed in cursor's value contains the child and you are not searching for a mutated element in a non-mutated Cursor value?                                                                                                                                                                                                                                                                                  
  org.openrewrite.java.service.SourcePositionService.positionOfChild(SourcePositionService.java:401)                                                                                                                                                                                                                      
  org.openrewrite.java.service.SourcePositionService.columnsOf(SourcePositionService.java:90)                                                                                                                                                                                                                             
  org.openrewrite.java.format.WrappingAndBracesVisitor.shouldWrap(WrappingAndBracesVisitor.java:529)                                                                                                                                                                                                                      
  org.openrewrite.java.format.WrappingAndBracesVisitor.lambda$visitContainer$1(WrappingAndBracesVisitor.java:89)                                                                                                                                                                                                          
  org.openrewrite.internal.ListUtils.map(ListUtils.java:206)                                                                                                                                                                                                                                                              
  org.openrewrite.java.format.WrappingAndBracesVisitor.visitContainer(WrappingAndBracesVisitor.java:84)                                                                                                                                                                                                                   
  org.openrewrite.java.JavaVisitor.visitTry(JavaVisitor.java:1177)                                                                                                                                                                                                                                                        
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:330)                                                                                                                                                                                                                                                   
  org.openrewrite.java.JavaIsoVisitor.visitTry(JavaIsoVisitor.java:30)                                                                                                                                                                                                                                                    
  org.openrewrite.java.tree.J$Try.acceptJava(J.java:5701)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.tree.J.accept(J.java:55)                                                                                                                                                                                                                                                                           
  org.openrewrite.TreeVisitor.visit(TreeVisitor.java:242)                                                                                                                                                                                                                                                                 
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:461)                                                                                                                                                                                                                           
  org.openrewrite.java.format.WrappingAndBracesVisitor.visit(WrappingAndBracesVisitor.java:48)                                                                                                                                                                                                                            
  org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:328)                                                                                                                                                                                                                                                          
  org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1350)                                                                                                                                                                                                                                                
  ...                                                                                                                                                                                                                                                                                                                     
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo$ResultsContainer$1.lambda$preVisit$0 (AbstractRewriteBaseRunMojo.java:395)
    at java.util.Optional.ifPresent (Optional.java:178)
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo$ResultsContainer$1.preVisit (AbstractRewriteBaseRunMojo.java:392)
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo$ResultsContainer$1.preVisit (AbstractRewriteBaseRunMojo.java:388)
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo_ResultsContainer_1_JavaVisitor.preVisit (AbstractRewriteBaseRunMojo_ResultsContainer_1_JavaVisitor.zig:81)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:239)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:328)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1350)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:426)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:245)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:269)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:425)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:833)
    at org.openrewrite.java.tree.J.accept (J.java:55)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:242)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:328)
    at org.openrewrite.java.JavaVisitor.visitMethodDeclaration (JavaVisitor.java:879)
    at org.openrewrite.java.tree.J$MethodDeclaration.acceptJava (J.java:4078)
    at org.openrewrite.java.tree.J.accept (J.java:55)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:242)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:328)
    at org.openrewrite.java.JavaVisitor.visitRightPadded (JavaVisitor.java:1350)
    at org.openrewrite.java.JavaVisitor.lambda$visitBlock$4 (JavaVisitor.java:426)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:245)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:269)
    at org.openrewrite.java.JavaVisitor.visitBlock (JavaVisitor.java:425)
    at org.openrewrite.java.tree.J$Block.acceptJava (J.java:833)
    at org.openrewrite.java.tree.J.accept (J.java:55)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:242)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:328)
    at org.openrewrite.java.JavaVisitor.visitClassDeclaration (JavaVisitor.java:508)
    at org.openrewrite.java.tree.J$ClassDeclaration.acceptJava (J.java:1382)
    at org.openrewrite.java.tree.J.accept (J.java:55)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:242)
    at org.openrewrite.TreeVisitor.visitAndCast (TreeVisitor.java:328)
    at org.openrewrite.java.JavaVisitor.lambda$visitCompilationUnit$9 (JavaVisitor.java:520)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:245)
    at org.openrewrite.internal.ListUtils.map (ListUtils.java:269)
    at org.openrewrite.java.JavaVisitor.visitCompilationUnit (JavaVisitor.java:520)
    at org.openrewrite.java.tree.J$CompilationUnit.acceptJava (J.java:1664)
    at org.openrewrite.java.tree.J.accept (J.java:55)
    at org.openrewrite.TreeVisitor.visit (TreeVisitor.java:242)
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo$ResultsContainer.getRecipeErrors (AbstractRewriteBaseRunMojo.java:399)
    at org.openrewrite.maven.AbstractRewriteBaseRunMojo$ResultsContainer.getFirstException (AbstractRewriteBaseRunMojo.java:379)
    at org.openrewrite.maven.AbstractRewriteDryRunMojo.execute (AbstractRewriteDryRunMojo.java:76)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:919)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:285)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:207)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java
@@ -1690,6 +1690,55 @@ class WrappingAndBracesTest implements RewriteTest {
 
     @Test
     @Issue("https://github.com/openrewrite/rewrite/issues/8073")
+    void tryWithResourcesChopIfTooLongWithBody() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withTryWithResources(wrap.getTryWithResources().withWrap(ChopIfTooLong).withAlignWhenMultiline(true))
+          ),
+          java(
+            """
+              import java.io.*;
+
+              class Test {
+                  void test() {
+                      try (FileReader fr = new FileReader("input.txt"); BufferedReader br = new BufferedReader(fr); FileWriter fw = new FileWriter("output.txt")) {
+                          String line = br.readLine();
+                          while (line != null) {
+                              fw.write(line);
+                              line = br.readLine();
+                          }
+                      } catch (IOException e) {
+                          throw new RuntimeException(e);
+                      }
+                  }
+              }
+              """,
+            """
+              import java.io.*;
+
+              class Test {
+                  void test() {
+                      try (FileReader fr = new FileReader("input.txt");
+                           BufferedReader br = new BufferedReader(fr);
+                           FileWriter fw = new FileWriter("output.txt")) {
+                          String line = br.readLine();
+                          while (line != null) {
+                              fw.write(line);
+                              line = br.readLine();
+                          }
+                      } catch (IOException e) {
+                          throw new RuntimeException(e);
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/8073")
     void tryWithResourcesChopIfTooLongNotLongEnough() {
         rewriteRun(
           autoFormat(

--- a/rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java
+++ b/rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java
@@ -1654,6 +1654,64 @@ class WrappingAndBracesTest implements RewriteTest {
     }
 
     @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/8073")
+    void tryWithResourcesChopIfTooLong() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withTryWithResources(wrap.getTryWithResources().withWrap(ChopIfTooLong).withAlignWhenMultiline(true))
+          ),
+          java(
+            """
+              import java.io.*;
+
+              class Test {
+                  void test() {
+                      try (FileReader fr = new FileReader("input.txt"); BufferedReader br = new BufferedReader(fr); FileWriter fw = new FileWriter("output.txt")) {
+                      }
+                  }
+              }
+              """,
+            """
+              import java.io.*;
+
+              class Test {
+                  void test() {
+                      try (FileReader fr = new FileReader("input.txt");
+                           BufferedReader br = new BufferedReader(fr);
+                           FileWriter fw = new FileWriter("output.txt")) {
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/issues/8073")
+    void tryWithResourcesChopIfTooLongNotLongEnough() {
+        rewriteRun(
+          autoFormat(
+            spaces -> spaces,
+            wrap -> wrap.withTryWithResources(wrap.getTryWithResources().withWrap(ChopIfTooLong).withAlignWhenMultiline(true))
+          ),
+          java(
+            """
+              import java.io.*;
+
+              class Test {
+                  void test() {
+                      try (FileReader fr = new FileReader("input.txt"); BufferedReader br = new BufferedReader(fr)) {
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
     void tryCatchOnNewLine() {
         rewriteRun(
           autoFormat(

--- a/rewrite-java/src/main/java/org/openrewrite/java/service/SourcePositionService.java
+++ b/rewrite-java/src/main/java/org/openrewrite/java/service/SourcePositionService.java
@@ -218,6 +218,39 @@ public class SourcePositionService {
             }
 
             @Override
+            public J visitTry(J.Try tryable, PrintOutputCapture<TreeVisitor<?, ?>> p) {
+                JContainer<J.Try.Resource> resources = tryable.getPadding().getResources();
+                // JavaPrinter#visitTry prints the resources directly rather than through visitContainer/visitRightPadded,
+                // so the generic findJContainer search never matches. Replicate that printing here while tracking the span.
+                if (findJContainer == null || resources == null ||
+                        (List<?>) resources.getPadding().getElements() != (List<?>) findJContainer.getPadding().getElements()) {
+                    return super.visitTry(tryable, p);
+                }
+                beforeSyntax(tryable, Space.Location.TRY_PREFIX, p);
+                p.append("try");
+                visitSpace(resources.getBefore(), Space.Location.TRY_RESOURCES, p);
+                p.append('(');
+                List<JRightPadded<J.Try.Resource>> elements = resources.getPadding().getElements();
+                for (int i = 0; i < elements.size(); i++) {
+                    JRightPadded<J.Try.Resource> paddedResource = elements.get(i);
+                    J.Try.Resource resource = paddedResource.getElement();
+                    if (i == 0) {
+                        resource = startFind(resource, p);
+                    } else {
+                        visitSpace(resource.getPrefix(), Space.Location.TRY_RESOURCE, p);
+                    }
+                    visitMarkers(resource.getMarkers(), p);
+                    visit(resource.getVariableDeclarations(), p);
+                    if (resource.isTerminatedWithSemicolon()) {
+                        p.append(';');
+                    }
+                    visitSpace(paddedResource.getAfter(), Space.Location.TRY_RESOURCE_SUFFIX, p);
+                }
+                found.set(true);
+                return tryable;
+            }
+
+            @Override
             protected void visitRightPadded(List<? extends JRightPadded<? extends J>> nodes, JRightPadded.Location location, String suffixBetween, PrintOutputCapture<TreeVisitor<?, ?>> p) {
                 if (findJContainer != null && nodes == findJContainer.getPadding().getElements()) {
                     for (int i = 0; i < nodes.size(); i++) {

